src/main/java/pedido/Pedido.java

```diff
@@ -1,19 +1,88 @@
 package pedido;
 
+import java.io.Serializable;
 import java.util.ArrayList;
+import java.util.Objects;
 
-public class Pedido{
 
-    private int id;
-    private  ArrayList<ItemPedido> itens;
-    private Cliente cliente;
+public class Pedido  {
 
-    public Pedido(int id, ArrayList<ItemPedido> itens,Cliente cliente){
+    private final int id;
+    private final ArrayList<ItemPedido> itens;
+    private final Cliente cliente;
+
+    public Pedido(int id, ArrayList<ItemPedido> itens, Cliente cliente){
         this.id = id;
         this.itens=itens;
         this.cliente=cliente;
     }
 
+    public double calcularTotal(Cardapio cardapio){
+        double total= 0;
+
+        for (ItemPedido item: itens) {
+            var shake = item.getShake();
+            int qtdShake = item.getQuantidade();
+            var adicionais = shake.getAdicionais();
+
+            var precoBase = cardapio.getPrecos().get(shake.getBase());
+            var precoBaseComTamanho = precoBase + (precoBase * shake.getTipoTamanho().multiplicador);
+            var precoComQuantidade = precoBaseComTamanho;
+            var totalAdicionais = adicionais.stream().map(adicional -> cardapio.getPrecos().get(adicional))
+                    .reduce(Double::sum).orElse(0.0);
+
+            total += (precoComQuantidade + totalAdicionais) *  qtdShake;
+        }
+
+        return total;
+    }
```

Comment:
Muito bom!!! O que você ainda poderia fazer é remover o for, utilizando um map e fazendo um reduce ao final. Vou anexar aqui a maneira como eu implementei. 
```
 public double calcularTotal(final Cardapio cardapio) {
        return itens.stream()
                .map(itemPedido -> {
                    final var shake = itemPedido.getShake();
                    final var adiocinais = shake.getAdicionais();
                    final var precoBase = cardapio.buscarPreco(shake.getBase());
                    final var precoAdicionais = adiocinais.stream().map(cardapio::buscarPreco).reduce(0.0, Double::sum);
                    final var acrescimoBase = shake.getTipoTamanho().multiplicador * precoBase;
                    return (precoBase + acrescimoBase + precoAdicionais) * itemPedido.getQuantidade();
                })
                .reduce(0.0, Double::sum);
    }
```